libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/push/DefaultPushHandler.kt

```diff
@@ -88,32 +90,35 @@ class DefaultPushHandler @Inject constructor(
-                                        roomId = request.roomId,
-                                        sessionId = request.sessionId,
-                                        reason = "Showing fallback notification",
-                                    )
-                                    mutableBatteryOptimizationStore.showBatteryOptimizationBanner()
-                                } else {
-                                    pushHistoryService.onSuccess(
-                                        providerInfo = request.providerInfo,
-                                        eventId = request.eventId,
-                                        roomId = request.roomId,
-                                        sessionId = request.sessionId,
-                                        comment = "Push handled successfully",
-                                    )
-                                }
+                                pushHistoryService.onSuccess(
+                                    providerInfo = request.providerInfo,
+                                    eventId = request.eventId,
+                                    roomId = request.roomId,
+                                    sessionId = request.sessionId,
+                                    comment = "Push handled successfully",
+                                )
                             },
                             onFailure = { exception ->
+                                val reason = when (exception) {
+                                    is NotificationResolverException.EventNotFound -> "Event not found"
+                                    is NotificationResolverException.EventFilteredOut -> {
+                                        pushHistoryService.onSuccess(
+                                            providerInfo = request.providerInfo,
+                                            eventId = request.eventId,
+                                            roomId = request.roomId,
+                                            sessionId = request.sessionId,
+                                            comment = "Push handled successfully but notification was filtered out",
+                                        )
+                                        return@fold
+                                    }
+                                    else -> "Unknown error: ${exception.message}"
+                                }
                                 pushHistoryService.onUnableToResolveEvent(
                                     providerInfo = request.providerInfo,
                                     eventId = request.eventId,
                                     roomId = request.roomId,
                                     sessionId = request.sessionId,
-                                    reason = exception.message ?: exception.javaClass.simpleName,
+                                    reason = "$reason - Showing fallback notification",
```

Comment:
Why adding "Showing fallback notification" to the reason?

Reply:
I think the users are already used to seeing this message, it would also be an explanation for them as to why the 'You have messages' notification is displayed. I can remove it if you think it doesn't add value.